psutil/_psutil_linux.c

```diff
@@ -298,29 +298,25 @@ psutil_proc_cpu_affinity_set(PyObject *self, PyObject *args) {
     pid_t pid;
     int i, seq_len;
     PyObject *py_cpu_set;
-    PyObject *py_cpu_seq = NULL;
 
     if (!PyArg_ParseTuple(args, _Py_PARSE_PID "O", &pid, &py_cpu_set))
         return NULL;
 
     if (!PySequence_Check(py_cpu_set)) {
-        PyErr_Format(PyExc_TypeError, "sequence argument expected, got %s",
-                     Py_TYPE(py_cpu_set)->tp_name);
+        PyErr_Format(PyExc_TypeError, "sequence argument expected, got %R", Py_TYPE(py_cpu_set));
         goto error;
     }
 
-    py_cpu_seq = PySequence_Fast(py_cpu_set, "expected a sequence or integer");
-    if (!py_cpu_seq)
-        goto error;
-    seq_len = PySequence_Fast_GET_SIZE(py_cpu_seq);
+    seq_len = PySequence_Size(py_cpu_set);
     CPU_ZERO(&cpu_set);
     for (i = 0; i < seq_len; i++) {
-        PyObject *item = PySequence_Fast_GET_ITEM(py_cpu_seq, i);
+        PyObject *item = PySequence_GetItem(py_cpu_set, i);
```

Comment:
```suggestion
        PyObject *item = PySequence_GetItem(py_cpu_set, i);
        if (!item) {
            goto error;
        }
```
(Previously, `PySequence_Fast` checked for errors while getting items.)